Pocket/pocket-ios
# feat(collections): add UI to saves and search cards

## Summary
Add Collection Label to Saves and Search Cards

## References 
IN-1595

## Implementation Details
* Added collection label to `ItemListItemCell` for Saves / Archive and `ItemDetails` for Search
* Add attributed string for `Collection` to `ItemsListItemPresenter` to be used for both views

## Test Steps
- [x] Given I am viewing my Saves,
when a saved item is a collection,
then the cell for that item should be updated to reflect that it is a collection.
- [x] Given I am viewing my Saves,
when a saved item is not a collection,
then the cell for that item should not be updated.
- [x] Given I am viewing my Archive,
when an archived item is a collection,
then the cell for that item should be updated to reflect that it is a collection.
- [x] Given I am viewing my Archive,
when an archived item is not a collection,
then the cell for that item should not be updated.

* Apply testing steps to Search as well for all the different scopes.

## PR Checklist:
- N / A Added Unit / UI tests

## Screenshots
| Archive | Saves | Search |
| --- | --- | --- |
| ![Simulator Screenshot - iPhone 14 Pro - 2023-07-28 at 11 32 25](https://github.com/Pocket/pocket-ios/assets/6743397/a028fc31-9108-43c2-8308-79f0cefdeaf3) | ![Simulator Screenshot - iPhone 14 Pro - 2023-07-28 at 11 32 09](https://github.com/Pocket/pocket-ios/assets/6743397/97146326-59ae-4c01-a840-ce74b4ec8252) | ![Simulator Screenshot - iPhone 14 Pro - 2023-07-28 at 11 32 59](https://github.com/Pocket/pocket-ios/assets/6743397/1bfca2e2-778f-4e71-8921-724f850da828)|
